Engine/src/Engine/ECS/SparseArray.h

```diff
@@ -0,0 +1,165 @@
+#pragma once
+#include "pch.h"
+#include "Engine/Memory/MemoryCommon.h"
+#include "ECS_Utility.h"
+
+namespace engine
+{
+    template<typename T, std::size_t MAX_SIZE, typename DenseAllocator = std::allocator<T>, 
+        typename SparseAllocator = std::allocator<std::size_t>>
+    class SparseContainer
+    {
+
+
+    private:
+        /*using alloc_type = typename std::allocator_traits<Allocator>::template rebind_alloc<T>;
```

Comment:
Delete away if unused?

Reply:
In the future, this class will likely be improved upon, especially in terms of memory allocation. So thats why it is there

Reply:
ah ok